Extensions/Multiplayer/multiplayertools.ts

```diff
@@ -1519,6 +1602,164 @@ namespace gdjs {
+          'The game ID is missing, the quick join lobby action cannot continue.'
+        );
+        return;
+      }
+
+      _quickJoinLobbyFailureReason = null;
+      _isQuickJoiningOrStartingAGame = true;
+      if (displayLoader) {
+        gdjs.multiplayerComponents.displayLoader(runtimeScene, true);
+      }
+
+      const quickJoinLobbyRelativeUrl = `/play/game/${_gameId}/public-lobby/action/quick-join`;
+      const platformInfo = runtimeScene.getGame().getPlatformInfo();
+
+      try {
+        const quickJoinLobbyResponse: QuickJoinLobbyResponse = await gdjs.evtTools.network.retryIfFailed(
+          { times: 2 },
+          () =>
+            fetchAsPlayer({
+              relativeUrl: quickJoinLobbyRelativeUrl,
+              method: 'POST',
+              dev: isUsingGDevelopDevelopmentEnvironment,
+              body: JSON.stringify({
+                isPreview: runtimeScene.getGame().isPreview(),
+                gameVersion: runtimeScene.getGame().getGameData().properties
+                  .version,
+                supportedCompressionMethods:
+                  platformInfo.supportedCompressionMethods,
+              }),
+            })
+        );
+
+        if (
+          quickJoinLobbyResponse.status === 'full' ||
+          quickJoinLobbyResponse.status === 'not-enough-players'
+        ) {
+          _quickJoinLobbyJustFailed = true;
+          _quickJoinLobbyFailureReason =
+            quickJoinLobbyResponse.status === 'full'
+              ? 'FULL'
+              : 'NOT_ENOUGH_PLAYERS';
+          onLobbyQuickJoinFinished(runtimeScene);
+          if (openLobbiesPageIfFailure) {
+            openLobbiesWindow(runtimeScene);
+          }
+          return;
+        }
+
+        if (quickJoinLobbyResponse.status === 'join-game') {
+          if (quickJoinLobbyResponse.lobby.status === 'waiting') {
+            _actionAfterJoiningLobby = 'START_GAME';
+          } else if (quickJoinLobbyResponse.lobby.status === 'playing') {
+            _actionAfterJoiningLobby = 'JOIN_GAME';
+          } else {
+            throw new Error(
+              `Lobby in wrong status: ${quickJoinLobbyResponse.status}`
+            );
+          }
+        } else {
+          if (_connectionId) {
+            // Already connected to a lobby.
+            onLobbyQuickJoinFinished(runtimeScene);
+            openLobbiesWindow(runtimeScene);
+            return;
+          } else {
+            _actionAfterJoiningLobby = 'OPEN_LOBBY_PAGE';
+          }
+        }
+        handleJoinLobbyEvent(runtimeScene, quickJoinLobbyResponse.lobby.id);
+      } catch (error) {
+        logger.error('An error occurred while joining a lobby:', error);
+        _quickJoinLobbyJustFailed = true;
+        _quickJoinLobbyFailureReason = 'UNKNOWN';
+        onLobbyQuickJoinFinished(runtimeScene);
+        if (openLobbiesPageIfFailure) {
+          openLobbiesWindow(runtimeScene);
+        }
+      }
+    };
+
+    export const quickJoinLobby = async (
```

Comment:
could almost be called `authenticateAndQuickJoinLobby`, so that the other method can be just called "quickJoinLobby" (rather than "doQuickJoinLobby", which makes me think "ok sure but why is there a need for this DO method? What is the other doing that this one is no doing?")